src/ldp.rs

```diff
@@ -615,6 +615,7 @@ async fn sign(
         created: Some(options.created.unwrap_or_else(now_ms)),
         domain: options.domain.clone(),
         challenge: options.challenge.clone(),
+        property_set: options.property_set.clone(),
```

Comment:
As mentioned in https://github.com/spruceid/ssi/pull/189/files#r661613315:

> I don't think `property_set` should be passed through into the proofs. There may be issue options that we don't support or know about but that are supposed to have a meaning other than passing them through into the proof. e.g. `credentialStatus` issue option is supposed to cause a `credentialStatus` object to be added to the document rather than to the proof: https://github.com/w3c-ccg/vc-http-api/blob/7c72ae2b470ff8dfec12d3b6c3de4f84c6f2a176/packages/vc-http-api-test-server/__tests__/issueCredential.spec.js#L130-L148

Maybe the options object with open-ended `property_set`, or the `property_set` alone, could be [transcoded](https://serde.rs/transcode.html) into a stricter version that is parameterized by the proof type and purpose (like in `feat/generic_proof`), so unknown properties would be caught at that point? Or it could be done non-generically, for the sake of just this PR.

Reply:
I experimented with generic `property_set`s for both `Proof` and `LinkedDataProofOptions` where the `property_set` type `P` for `Proof<P>` must implement `From<&LinkedDataProofOptions<O>>` (so that devs can implement how the options translate to custom proof fields). This path is blocked by some custom proof logic (e.g. `P256BLAKE2BDigest...2021` expects to add the `publicKeyJwk` field to the proof properties). Will try a non-generic path.

Reply:
`property_set` has been removed from `LinkedDataProofOptions` in favour of changing the `ProofSuite` trait to accept an additional `extra_proof_properties: Option<Map<String, Value>>` argument for `sign` and `prepare`.